app/controllers/application_controller.rb

```diff
@@ -10,4 +10,10 @@ def load_featured_products
   def cart
     @cart ||= Cart.new(session[:cart])
   end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
Should we add an instance variable assignment with memoization on this line for the server not to have to run this query every single time? What do you all think?